src/components/PressableWithSecondaryInteraction/index.js

```diff
@@ -1,106 +1,128 @@
+             */
+            if (withoutFocusOnSecondaryInteraction && pressableRef && pressableRef.current) {
+                pressableRef.current.blur();
+            }
+        },
+        [onSecondaryInteraction, preventDefaultContextMenu, pressableRef, withoutFocusOnSecondaryInteraction],
+    );
+
+    useEffect(() => {
+        if (!pressableRef || !pressableRef.current) {
             return;
         }
 
-        e.stopPropagation();
-        if (this.props.preventDefaultContextMenu) {
-            e.preventDefault();
+        if (innerRef) {
+            if (_.isFunction(innerRef)) {
+                innerRef(pressableRef);
+            } else if (_.isObject(innerRef)) {
+                innerRef.current = pressableRef.current;
+            }
         }
 
-        this.props.onSecondaryInteraction(e);
-        /**
-         * This component prevents the tapped element from capturing focus.
-         * We need to blur this element when clicked as it opens modal that implements focus-trapping.
-         * When the modal is closed it focuses back to the last active element.
-         * Therefore it shifts the element to bring it back to focus.
-         * https://github.com/Expensify/App/issues/14148
-         */
-        if (this.props.withoutFocusOnSecondaryInteraction && this.pressableRef) {
-            this.pressableRef.blur();
-        }
-    }
-
-    render() {
-        const defaultPressableProps = _.omit(this.props, ['onSecondaryInteraction', 'children', 'onLongPress']);
-        const inlineStyle = this.props.inline ? styles.dInline : {};
-
-        // On Web, Text does not support LongPress events thus manage inline mode with styling instead of using Text.
-        return (
-            <PressableWithFeedback
-                wrapperStyle={StyleUtils.combineStyles(DeviceCapabilities.canUseTouchScreen() ? [styles.userSelectNone, styles.noSelect] : [], inlineStyle)}
-                onPressIn={this.props.onPressIn}
-                onLongPress={this.props.onSecondaryInteraction ? this.executeSecondaryInteraction : undefined}
-                pressDimmingValue={this.props.activeOpacity}
-                onPressOut={this.props.onPressOut}
-                onPress={this.props.onPress}
-                ref={(el) => (this.pressableRef = el)}
-                // eslint-disable-next-line react/jsx-props-no-spreading
-                {...defaultPressableProps}
-                style={(state) => [StyleUtils.parseStyleFromFunction(this.props.style, state), inlineStyle]}
-            >
-                {this.props.children}
-            </PressableWithFeedback>
-        );
-    }
+        const element = pressableRef.current;
+        element.addEventListener('contextmenu', executeSecondaryInteractionOnContextMenu);
+
+        return () => {
+            element.removeEventListener('contextmenu', executeSecondaryInteractionOnContextMenu);
+        };
+    }, [executeSecondaryInteractionOnContextMenu, innerRef, pressableRef]);
```

Comment:
You converted `componentDidMount()` to this `useEffect`, its dependency array should be empty. If you need to disable the ESLint It's fine to do it as long you provide an explanation for that -> https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#what-is-the-exhaustive-deps-lint-rule-can-i-ignore-it

Reply:
You can also put the whole `executeSecondaryInteractionOnContextMenu` function inside the `useEffect`, since it's only used here.